Kentor.AuthServices/Configuration/SPOptions.cs

```diff
@@ -217,9 +220,107 @@ public IdentityConfiguration SystemIdentityModelIdentityConfiguration
             }
         }
 
+        readonly Collection<ServiceCertificate> serviceCertificates = new Collection<ServiceCertificate>();
+
+        /// <summary>
+        /// Certificates used by the service provider for signing or decryption.
+        /// </summary>
+        public Collection<ServiceCertificate> ServiceCertificates
+        {
+            get
+            {
+                return serviceCertificates;
+            }
+        }
+
+        /// <summary>
+        /// Certificates valid for use in decryption
+        /// </summary>
+        public ReadOnlyCollection<X509Certificate2> DecryptionServiceCertificates
+        {
+            get
+            {
+                var decryptionCertificates = ServiceCertificates
+                    .Where(c => c.Use == CertificateUse.Encryption || c.Use == CertificateUse.Both)
+                    .Select(c => c.Certificate);
+
+                return decryptionCertificates.ToList().AsReadOnly();
+            }
+        }
+
+        /// <summary>
+        /// Certificate for use in signing outbound requests
+        /// </summary>
+        public X509Certificate2 SigningServiceCertificate
+        {
+            get
+            {
+                var decryptionCertificates = ServiceCertificates
+                    .Where(c => c.Status == CertificateStatus.Current)
+                    .Where(c => c.Use == CertificateUse.Signing || c.Use == CertificateUse.Both)
+                    .Select(c => c.Certificate);
+
+                return decryptionCertificates.FirstOrDefault();
+            }
+        }
+
         /// <summary>
-        /// Certificate for service provider to use when decrypting assertions
+        /// Certificates to be published in metadata
         /// </summary>
-        public X509Certificate2 ServiceCertificate { get; set; }
+        public ReadOnlyCollection<ServiceCertificate> MetadataCertificates
+        {
+            get
+            {
+                var futureEncryptionCertExists = publishableServiceCertificates
+                    .Any(c => c.Status == CertificateStatus.Future && c.Use == CertificateUse.Encryption);
```

Comment:
You're right. Fixed and added test `SPOptions_MetadataCertificates_OnlyOneEncryptionPublished_FutureBoth`